### ISSUE TITLE
HTTP 2 support

### ISSUE THREAD

**Issue by dodekeract** [spirit-js/spirit#12]
Is there any HTTP 2 support for spirit? Is it possible to implement this with the current middleware API?

Combining resources would be the most important bit for my use-case.

**Comment by Lichtjaeger** [comment]
I would think this is not spirit related.

Use the [http2](https://www.npmjs.com/package/http2) module instead of the http module.

**Comment by dodekeract** [comment]
@Lichtjaeger Yes, it might be possible to "hack" a solution that'll work by including `['request']` in every route. However that'd completely remove the main advantages of spirit, namely the easy-to-write unit-tests and the verbose, but precise syntax. Something like `return {body, header, status, push}` would be a lot nicer.

**Comment by hnry** [comment]
Was looking at http2 support very early on but the module was buggy so I left it. 

So what you proposed @dodekeract was my initial plan. It'll have to go in the node adapter. 

The middleware space would be used for higher level support if someone wants to do something more automated or specific to their needs. 

I'll need some time to review http2 module and spec.

**Comment by dodekeract** [comment]
I'm fine with waiting, since we can (for the moment) just use CloudFlare, which supports HTTP2 by setting some custom headers.

**Comment by dodekeract** [comment]
@hnry https://medium.com/@jasnell/say-hello-to-http-2-for-node-js-core-261ba493846e

Anything we can do so far? I think being one of the first HTTP2 frameworks could do wonders for popularity of spirit.
